crates/core/tedge_agent/src/restart_manager/actor.rs

```diff
@@ -170,16 +170,13 @@ impl RestartManagerActor {
             }
             Err(err) => {
                 match err {
-                    StateError::FromIo(e) => match e.kind() {
-                        std::io::ErrorKind::NotFound => {
-                            info!("tedge-agent state file is not present")
-                        }
-                        std::io::ErrorKind::PermissionDenied => {
-                            error!("Failed to read tedge-agent state file due to: PermissionDenied")
-                        }
-                        e => error!("Failed to read tedge-agent state file due to: {e}"),
-                    },
-                    e => error!("Failed to read tedge-agent state file due to: {e}"),
+                    // file missing is not an error
+                    StateError::LoadingFromFileFailed { path, source }
+                        if source.kind() == std::io::ErrorKind::NotFound =>
+                    {
+                        info!("tedge-agent state file `{path}` is not present");
```

Comment:
```suggestion
                        info!("No pending restart operations found");
```
From the user's perspective, that's what the missing file means. I'm even okay not to log anything for this case, as the `process_pending_sm_operation` doesn't log this case either. It doesn't log the other serious IO errors like `PermissionDenied` either, but those are handled differently.

Reply:
> From the user's perspective, that's what the missing file means. I'm even okay not to log anything for this case, as the `process_pending_sm_operation` doesn't log this case either.

Okay, now I understand it a bit better, I agree it makes little sense to log it.

> It doesn't log the other serious IO errors like `PermissionDenied` either, but those are handled differently.

Can you expand on that? To me it looks like `process_pending_sm_operation` creates error responses when file could be read, but the state itself is inconsistent, so io-related errors are just discarded.

Reply:
> Can you expand on that? To me it looks like process_pending_sm_operation creates error responses when file could be read, but the state itself is inconsistent, so io-related errors are just discarded.

In the `process_pending_sm_operation` function, the `self.state_repository.load()` call is immediately followed by `state_repository.clear()`. So, even though the `load()` returns an IO error which is ignored, the very next `clear()` would also fail and that error would be returned. That's what I meant by "handled differently". Ideally, the error from `load` itself should have been returned.